src/components/chat/ChatApp.js

```diff
@@ -0,0 +1,108 @@
+// TODO: convert to TypeScript once this PR has been merged https://github.com/pusher/chatkit-client-js/pull/180
+import { ChatManager, TokenProvider } from '@pusher/chatkit-client';
+import jwt_decode from 'jwt-decode';
+import * as React from 'react';
+import Input from './Input.tsx';
+import MessageList from './MessageList.tsx';
+import { BACKEND_URL, INSTANCE_LOCATOR } from '../../utils/constants';
+import { IState } from '../../reducers/states';
+
+class ChatApp extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      connected: false,
+      currentRoom: {},
+      currentUser: {},
+      messages: []
+    };
+    this.addMessage = this.addMessage.bind(this);
+  }
+  /*
+  To keep the chat view at its bottom (and also where the input field is) we create a dummy div at the bottom.
+  It will be automatically rendered and scrolled to everytime the chat is updated.
+  */
+  messagesEndRef = React.createRef(); // for scrolling
+
+  componentDidUpdate() {
+    if (this.state.connected) {
+      this.scrollToBottom();
+    }
+  } // for scrolling
+
+  scrollToBottom = () => {
+    this.messagesEndRef.current.scrollIntoView({ behavior: 'smooth' });
+  };
+
+  componentDidMount() {
+    // tslint:disable-next-line:no-console
+    console.log(this.props.roomId);
```

Comment:
why is this committed?

Reply:
This helps the chat auto scroll to bottom everytime it is updated.

Elsewise we need to manually scroll down.

Reply:
This helps the chat auto scroll to bottom everytime it is updated.

Elsewise we need to manually scroll down.

Reply:
the `console.log`?

Reply:
Ah that one.

Aihui was looking at this issue: updating grade causes comment to be stored as undefined. I think she was debugging.

Reply:
I will fix this no worries!

Reply:
I've removed it sorry about that!

Reply:
No problem :) Thanks for helping to clean this up